azurestack/resource_arm_subnet.go

```diff
@@ -238,6 +237,35 @@ func resourceArmSubnetDelete(d *schema.ResourceData, meta interface{}) error {
 
 		azureStackLockByName(routeTableName, routeTableResourceName)
 		defer azureStackUnlockByName(routeTableName, routeTableResourceName)
+
+		// Get the subnet to dissasociate the route table, if we don't do this
```

Comment:
This is different then the AzureRM code. Is this logic required for stack and not AzureRM? Could we get a test for this behaviour?